src/AuthContext.tsx

```diff
@@ -137,6 +138,7 @@ export const AuthProvider = ({ authConfig, children }: IAuthProvider) => {
               if (initial) login()
             }
           })
+        setRefreshInProgress(false)
```

Comment:
this should probably happen in a finally block of the `fetchWithRefreshToken` promise if I'm not mistaken? 
Otherwise, it's resetting to false before the response of the token refresh has been ben received from my understanding.

Independently of that, if the tab ever gets closed / crashes while refreshing the token, `refreshInProgress` will never be reset to false which essentially breaks refreshing for all tabs for good until the user fully clears the browser cache?

Reply:
1: Very good point. 

2: Fair point. Refreshing will still fix this, as I added a "skip check" on `initial`, but this will botch the state if you have two tabs open and the one running the refresh is closed during the refreshing. This will only be fixed by refreshing/opening a new tab. I'd welcome a good alternative to this approach, as I realize it's a tad dirty.

Reply:
I suggest "aborted refreshRequest" and "session expired multiple instances" to be dealt with in separate issues/PRs.
This fix will likely solve 99% of problems that can arise with multiple tabs.